rxandroid/src/main/java/io/reactivex/android/schedulers/HandlerScheduler.java

```diff
@@ -0,0 +1,176 @@
+            return schedule(run, delay, period, unit);
+        }
+
+        private Disposable schedule(Runnable run, long initialDelay, long period, TimeUnit unit) {
+            if (run == null) throw new NullPointerException("run == null");
+            if (initialDelay < 0) throw new IllegalArgumentException("delay < 0: " + initialDelay);
+            if (unit == null) throw new NullPointerException("unit == null");
+
+            if (disposed) {
+                return Disposables.disposed();
+            }
+
+            run = RxJavaPlugins.onSchedule(run);
+
+            ScheduledRunnable scheduled = new ScheduledRunnable(handler, run, period, unit);
+
+            Message message = Message.obtain(handler, scheduled);
+            message.obj = this; // Used as token for batch disposal of this worker's runnables.
+
+            handler.sendMessageDelayed(message, unit.toMillis(initialDelay));
+
+            // Re-check disposed state for removing in case we were racing a call to dispose().
+            if (disposed) {
+                handler.removeCallbacks(scheduled);
+                return Disposables.disposed();
+            }
+
+            return scheduled;
+        }
+
+        @Override
+        public void dispose() {
+            disposed = true;
+            handler.removeCallbacksAndMessages(this /* token */);
+        }
+
+        @Override
+        public boolean isDisposed() {
+            return disposed;
+        }
+    }
+
+    private static final class ScheduledRunnable implements Runnable, Disposable {
+        private final Handler handler;
+        private final Runnable delegate;
+        /** The time delay for rescheduling or {@link #NO_PERIOD} if a one-shot execution. */
+        private final long periodMillis;
+
+        private volatile boolean disposed;
+
+        ScheduledRunnable(Handler handler, Runnable delegate, long period, TimeUnit unit) {
+            this.handler = handler;
+            this.delegate = delegate;
+            this.periodMillis = period == NO_PERIOD ? NO_PERIOD : unit.toMillis(period);
+        }
+
+        @Override
+        public void run() {
+            try {
+                delegate.run();
+            } catch (Throwable t) {
+                IllegalStateException ie;
+                if (t instanceof OnErrorNotImplementedException) {
+                    ie = new IllegalStateException(
+                        "Exception thrown on Scheduler. Add `onError` handling.", t);
+                } else {
+                    ie = new IllegalStateException("Fatal Exception thrown on Scheduler.", t);
+                }
+                RxJavaPlugins.onError(ie);
+                Thread thread = Thread.currentThread();
+                thread.getUncaughtExceptionHandler().uncaughtException(thread, ie);
+                return;
+            }
+
+            if (periodMillis != NO_PERIOD && !disposed) {
+                handler.postDelayed(this, periodMillis);
```

Comment:
Good spot. The only reason I implemented this here instead of delegating to the implementations already in `Scheduler` was that `schedulePeriodicallyDirect` has to create a `Worker` and ends up double-wrapping the `Runnable`. I wanted to have the minimal overhead.